File: src/main/java/net/snipsniper/utils/Function.java
package net.snipsniper.utils;

public abstract class Function {
    public boolean run() { return true; }
    public boolean run(String... args) { return true; }
    public boolean run(Object... args) { return true; }
    public boolean run(Boolean... args) { return true; }
    public boolean run(Integer... args) { return true; }
    public boolean run(ConfigSaveButtonState state) { return true; }
}

File: src/main/kotlin/net/snipsniper/utils/Function.kt
package net.snipsniper.utils

abstract class Function {
    open fun run(): Boolean = true
    open fun run(vararg args: String): Boolean = true
    //open fun run(vararg args: Any?): Boolean = true
    open fun run(vararg args: Boolean): Boolean = true
    open fun run(vararg args: Int): Boolean = true //TODO: Change to Int after portjob is complete
    open fun run(vararg args: ConfigSaveButtonState): Boolean = true
    open fun run(state: ConfigSaveButtonState): Boolean = true
}



fun baba(action: (String) -> (String)) {
    println("Action invoked: ${action.invoke("baba")}")
}

fun main() {
    baba {
        return@baba "hello"
    }
}